### PR TITLE
[results] Implement v1alpha2 ListResults.

### PR DESCRIPTION
**Pull request by wlynch**
# Changes

Largely caries over from v1alpha1 impl, refactoring common libraries.
Notable changes:

- Refactors pagination batch sizing and token encoding into its own library
to share between versions.
- Moves internal pagination proto out of the API package into its own library.
- Refactors CEL environment creation / parsing into its own library to
share between versions.
- Adds additional test coverage for refactored libraries.
- Adds test db logger which prints out any SQL statements made during
tests to t.Log.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [n/a] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Commit messages includes a project tag in the subject - e.g. [OCI], [hub], [results], [task-loops]

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by wlynch** [comment]
@XinruZhang If you have a chance, I'd appreciate your review here to make sure I did everything right w.r.t. pagination.

**Review by XinruZhang — COMMENTED** [review]
Only one comment on the pagination, the others lgtm :)

**Comment by ImJasonH** [comment]
Why bother versioning the API, instead of just making a breaking change? It's alpha and Experimental and has no real usage.

File: results/pkg/api/server/db/pagination/pagination.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	return pi.GetName(), pi.GetFilter(), err
+}
+
+// Batcher suggests dynamic batch sizes for list queries.
+type Batcher struct {
+	// Total number of items we want to fetch
+	want int
+
+	// Min/Max thresholds of number of items to fetch for a given batch.
+	min, max int
```
[the remark this answers is not in the document]
Comment (wlynch) [reply results/pkg/api/server/db/pagination/pagination.go:49]: Good call! Done.

This also uncovered a bug if the last entry in a batch was a match, which I fixed up.

**Comment by wlynch** [comment]
> Why bother versioning the API, instead of just making a breaking change? It's alpha and Experimental and has no real usage.

It makes it easier to move / rewrite pieces in smaller chunks and defers needing to modify the existing tests / watcher client. The plan is to remove v1alpha1 ASAP as soon as v1alpha2 is ready.

**Comment by wlynch** [comment]
Closing this in favor of https://github.com/tektoncd/results/pull/3

**Pull request closed without merging**